codegenerator/cli/templates/static/codegen/src/db/DbFunctionsImplementation.js

```diff
@@ -7,10 +7,10 @@ const chunkBatchQuery = async (sql, entityDataArray, queryToExecute) => {
   // Split entityDataArray into chunks of MAX_ITEMS_PER_QUERY
   for (let i = 0; i < entityDataArray.length; i += MAX_ITEMS_PER_QUERY) {
     const chunk = entityDataArray.slice(i, i + MAX_ITEMS_PER_QUERY);
-    const response = await queryToExecute(sql, chunk);
-    responses.push(response);
+    const pendingRes = queryToExecute(sql, chunk);
+    responses.push(pendingRes);
   }
-  return responses;
+  return Promise.all(responses);
```

Comment:
This had no effect before btw since the entity operations were done last. It now makes a difference since we need to cleanup after all the chunks are finished.